plugins/providers/virtualbox/driver/version_7_0.rb

```diff
@@ -61,6 +61,110 @@ def read_forwarded_ports(uuid=nil, active_only=false)
+        def create_host_only_network(options)
+          # Create the interface
+          execute("hostonlynets", "create", retryable: true) =~ /^Interface '(.+?)' was successfully created$/
+          name = $1.to_s
+
+          # Get the IP so we can determine v4 vs v6
+          ip = IPAddr.new(options[:adapter_ip])
+
+          # Configure
+          if ip.ipv4?
+            execute("hostonlynets", "ipconfig", name,
+                    "--ip", options[:adapter_ip],
+                    "--netmask", options[:netmask],
+                    retryable: true)
+          elsif ip.ipv6?
+            execute("hostonlynets", "ipconfig", name,
+                    "--ipv6", options[:adapter_ip],
+                    "--netmasklengthv6", options[:netmask].to_s,
+                    retryable: true)
+          else
+            raise "BUG: Unknown IP type: #{ip.inspect}"
+          end
+
+          # Return the details
+          return {
+            name: name,
+            ip:   options[:adapter_ip],
+            netmask: options[:netmask],
+            dhcp: nil
+          }
+        end
+        
+        def delete_unused_host_only_networks
+          networks = []
+          execute("list", "hostonlynets", retryable: true).split("\n").each do |line|
+            networks << $1.to_s if line =~ /^Name:\s+(.+?)$/
+          end
+
+          execute("list", "vms", retryable: true).split("\n").each do |line|
+            if line =~ /^".+?"\s+\{(.+?)\}$/
+              begin
+                info = execute("showvminfo", $1.to_s, "--machinereadable", retryable: true)
+                info.split("\n").each do |inner_line|
+                  if inner_line =~ /^hostonlynetwork\d+="(.+?)"$/
+                    networks.delete($1.to_s)
+                  end
+                end
+              rescue Vagrant::Errors::VBoxManageError => e
+                raise if !e.extra_data[:stderr].include?("VBOX_E_OBJECT_NOT_FOUND")
+
+                # VirtualBox could not find the vm. It may have been deleted
+                # by another process after we called 'vboxmanage list vms'? Ignore this error.
+              end
+            end
+          end
+
+          networks.each do |name|
+            # First try to remove any DHCP servers attached. We use `raw` because
+            # it is okay if this fails. It usually means that a DHCP server was
+            # never attached.
+            raw("dhcpserver", "remove", "--ifname", name)
+
+            # Delete the actual host only network interface.
+            execute("hostonlynets", "remove", name, retryable: true)
+          end
+        end
+        
+        def read_host_only_interfaces
+          execute("list", "hostonlynets", retryable: true).split("\n\n").collect do |block|
+            info = {}
+
+            block.split("\n").each do |line|
+              if line =~ /^Name:\s+(.+?)$/
+                info[:name] = $1.to_s
+              elsif line =~ /^IPAddress:\s+(.+?)$/
```

Comment:
This condition will never match, since this line doesn't exist in the output, at least for beta 3:

```console
❯ VBoxManage list hostonlynets
Name:            Legacy vboxnet0 Network
GUID:            50c1b470-b0e7-4900-bd56-26d9c4f9f5e2

State:           Enabled
NetworkMask:     255.255.255.0
LowerIP:         192.168.56.10
UpperIP:         192.168.56.254
VBoxNetworkName: hostonly-Legacy vboxnet0 Network
```

Reply:
was hoping it would be as simple as changing out the function from the previous command; need to dig a little deeper to get the network behavior over single ip.  Thanks for pointing that out

Reply:
Yeah I'm in a similar boat and need to dig into this further. I think the semantics have changed from listing network interfaces to listing networks, so we might need to make some structural changes here (maybe similar to the way `virtualbox__intnet` configures internal-only networks).

The VirtualBox driver feels so version-specific and fragile :/